tests/http/models/test_genric_resource.py

```diff
@@ -11,100 +9,43 @@ def meta_data():
-
-        resource = GenericResource(
-            data="test_data",
-            **{"$meta": meta_data},  # noqa: WPS445 WPS517
-        )
-
-        assert resource.data == "test_data"
-        assert resource._meta == meta_object
+def test_generic_resource_empty():
+    resource = GenericResource()
+    assert resource.meta is None
+    assert resource.to_dict() == {}
 
-    def test_dynamic_attribute_access(self):
-        resource = GenericResource()
 
-        resource.dynamic_field = "dynamic_value"
-        resource.nested_object = {"inner": "data"}
+def test_from_response(meta_data):
+    record_data = {"id": 1, "name": {"given": "Albert", "family": "Einstein"}}
+    response = Response(200, json={"data": record_data, "$meta": meta_data})
+    expected_meta = Meta.from_response(response)
 
-        assert resource.dynamic_field == "dynamic_value"
-        assert resource.nested_object.inner == "data"
+    resource = GenericResource.from_response(response)
 
+    assert resource.to_dict() == record_data
+    assert resource.meta == expected_meta
 
-class TestGenericResourceFromResponse:
-    @pytest.fixture
-    def meta_data_single(self):
-        return {"ignored": ["one"]}  # noqa: WPS226
 
-    @pytest.fixture
-    def meta_data_two_resources(self):
-        return {"pagination": {"limit": 10, "offset": 0, "total": 2}, "ignored": ["one"]}  # noqa: WPS226
+def test_attribute_getter(mocker, meta_data):
+    resource_data = {"id": 1, "name": {"given": "Albert", "family": "Einstein"}}
+    response = Response(200, json={"data": resource_data, "$meta": meta_data})
+    resource = GenericResource.from_response(response)
 
-    @pytest.fixture
-    def meta_data_multiple(self):
-        return {"ignored": ["one", "two"]}  # noqa: WPS226
+    assert resource.id == 1
+    assert resource.name.given == "Albert"
 
-    @pytest.fixture
-    def single_resource_data(self):
-        return {"id": 1, "name": "test"}
 
-    @pytest.fixture
-    def single_resource_response(self, single_resource_data, meta_data_single):
-        return Response(200, json={"data": single_resource_data, "$meta": meta_data_single})
+def test_attribute_setter():
+    resource_data = {"id": 1, "name": {"given": "Albert", "family": "Einstein"}}
+    resource = GenericResource(resource_data)
 
-    @pytest.fixture
-    def multiple_resource_response(self, single_resource_data, meta_data_two_resources):
-        return Response(
-            200,
-            json={
-                "data": [single_resource_data, single_resource_data],
-                "$meta": meta_data_two_resources,
-            },
-        )
+    resource.id = 2
+    assert resource.id == 2
 
-    def test_malformed_meta_response(self):
-        with pytest.raises(TypeError, match=re.escape("Response $meta must be a dict.")):
-            _resource = GenericResource.from_response(Response(200, json={"data": {}, "$meta": 4}))
+    resource.name.given = "John"
+    assert resource.name.given == "John"
```

Comment:
We are following AAA which means each section should be grouped together. Sorry, I can't suggest the change directly here.

```
resource.id = 2
resource.name.given = "John"

assert resource.id == 2
assert resource.name.given == "John"
```